filecoin-proofs/src/bin/paramfetch.rs

```diff
@@ -31,32 +41,46 @@ Set $FILECOIN_PARAMETER_CACHE to specify parameter directory. Defaults to '{}'
         )
         .get_matches();
 
+    let json = PathBuf::from(matches.value_of("json").unwrap_or("./parameters.json"));
+    let parameter_map = load_parameter_map(&json).expect(ERROR_PARAMETERS_MAPPED);
+
     if let Some(matches) = matches.subcommand_matches("fetch") {
         let parameters = if matches.is_present("all") {
-            get_mapped_parameters()
+            get_mapped_parameters(&parameter_map)
         } else {
-            choose_mapped_parameters()
+            choose_mapped_parameters(&parameter_map)
         }
         .expect(ERROR_PARAMETERS_MAPPED);
 
         if parameters.len() > 0 {
-            println!("fetching parameters:");
+            println!("fetching parameters");
+
+            for parameter in parameters.iter() {
+                println!("fetching '{}'...", parameter);
 
-            parameters.iter().for_each(|p| {
-                println!("{}...", p);
+                match fetch_parameter_file(&parameter_map, parameter.to_string()) {
+                    Ok(_) => {
+                        println!("checking digest...");
+                        let digest =
+                            get_parameter_digest(parameter.to_string()).expect(ERROR_DIGEST);
 
-                match fetch_parameter_file(p.to_string()) {
-                    Ok(_) => println!("ok"),
+                        if digest != parameter_map.get(parameter).unwrap().digest {
+                            println!("failed integrity check. you should remove this parameter and try again");
```

Comment:
I would consider an intermediate solution here of at least renaming the already downloaded (invalid) parameters file (appending something like `-invalid-digest`) to make it explicit that they don't match the digest (since a `println` can be easily overlooked). That way any posterior script that was waiting for a specific file would fail and would force the user to manually rename the invalid one.